force-app/main/default/classes/SOQL.cls

```diff
@@ -345,17 +395,29 @@ public with sharing class SOQL {
      */
     public static void setPrimaryContact() {
         //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
-
+        List<Opportunity> getClosedWonOpp = getOpportunityDetailsForClosedWon();
         // Create a set of Account Ids from the Opportunities
-
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity currentOpp : getClosedWonOpp) {
+            accountIds.add(currentOpp.AccountId);
+        }
         // Query the Contacts
-
+        List<Contact> contactsList = [SELECT Id, AccountId FROM Contact WHERE AccountId IN :accountIds ORDER BY FirstName];
+        Map<Id,Id> accountToContactMap = new Map<Id,Id>();
         // Loop through each Contact and add the Account Id and Contact to the map
-        
+        Id previousAccountId;
```

Comment:
What are we trying to do here? What if the are hundreds of Account IDs -- and they're not ordered? Would your code still work the way you intended?